src/LanguageServer/Protocol/Features/Options/FallbackAnalyzerConfigOptionsProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Options;
+
+[Shared]
+[ExportWorkspaceService(typeof(IFallbackAnalyzerConfigOptionsProvider), workspaceKinds:
+    [WorkspaceKind.Host, WorkspaceKind.Interactive, WorkspaceKind.SemanticSearch, WorkspaceKind.MetadataAsSource, WorkspaceKind.MiscellaneousFiles, WorkspaceKind.Debugger, WorkspaceKind.Preview])]
```

Comment:
what workspaces woudl this *not* be exported for?  Do we need all the kinds?  or can we just export this and have it take over from whatever default (or no default) might be registered.

Reply:
Remote workspaces shouldn't have this.

Reply:
makes sense.  can you doc that explicitly?

Reply:
Once https://github.com/dotnet/roslyn/pull/74171 is in, we can replace this list with flags enum value "ClientWorkspaces" or something similar.

Reply:
taht works for me.